src/env.cpp

```diff
@@ -38,14 +38,6 @@
 #include "wcstringutil.h"
 #include "wutil.h"  // IWYU pragma: keep
 
-/// Some configuration path environment variables.
-#define FISH_DATADIR_VAR L"__fish_data_dir"
-#define FISH_SYSCONFDIR_VAR L"__fish_sysconf_dir"
-#define FISH_HELPDIR_VAR L"__fish_help_dir"
-#define FISH_BIN_DIR L"__fish_bin_dir"
-#define FISH_CONFIG_DIR L"__fish_config_dir"
-#define FISH_USER_DATA_DIR L"__fish_user_data_dir"
-
 /// At init, we read all the environment variables from this array.
 extern char **environ;
 
```

Comment:
yeah `key` makes no sense, it should be `key_and_val`. Testament to how this code was dead in practice and we are better of skipping such entries

Reply:
Note: This code is pretty new, and the "inheriteds" mechanism is there for one reason: To have `set --show foo` show `$foo: originally inherited as |banana|`. See #9029.

Reply:
I am going to not fix this now — `inheriteds` will not be the same on Rust vs Cpp due to the change in how we read env-vars (Rust allows starting with `=`, that turns into an empty name in our case, which is malformed).
I think that difference is is fine until `builtins/set` is ported